pandas/_libs/src/parser/tokenizer.c

```diff
@@ -1828,14 +1823,14 @@ double round_trip(const char *p, char **q, char decimal, char sci, char tsep,
     if (endpc == pc + strlen(pc)) {
         if (q != NULL) {
            // report endptr from source string (p)
-            *q = (char *) endptr;
+            *q = endptr;
         }
     } else {
         *error = -1;
         if (q != NULL) {
            // p and pc are different len due to tsep removal. Can't report
            // how much it has consumed of p. Just rewind to beginning.
-            *q = (char *)p;
+	  *q = (char *)p;  // TODO: this could be undefined behavior
```

Comment:
We probably just need better types here. I don't think `q` is ever actually mutated, but if it is this would be undefined behavior as the contents of `p` are qualified as const